### PR TITLE
Reworked user.die function to include deathreports and explosion trigger

### PR DESCRIPTION
**Pull request by Froggg**
changed user.die() into a method to not only kill the user but return all necessary responses for death (explosion recounts, death reports, any new ones in the future) and reworked all uses of .die to use the new method. cleaned out the now out of date code for creating deathreports.

### CODE REVIEW

File: ewcmd.py [hunk 1]
```diff
@@ -1018,12 +1015,11 @@ def weather_txt(id_server):
...
 
 
-		targetmodel.die(cause = ewcfg.cause_cliff)
+		die_resp = targetmodel.die(cause = ewcfg.cause_cliff)
 		targetmodel.persist()
 		await ewrolemgr.updateRoles(client=cmd.client, member=target)
-		if deathreport != "":
-			sewerchannel = ewutils.get_channel(cmd.message.server, ewcfg.channel_sewers)
-			await ewutils.send_message(cmd.client, sewerchannel, deathreport)
+		if die_resp != ewutils.EwResponseContainer(id_server = cmd.message.server.id_server):
```
Comment (magicksRampage) [inline ewcmd.py]: this check is unnecessary and doesn't work. if a response container is empty, it just won't post anything on .post(), and you can't compare objects like that

File: ewutils.py [hunk 2]
```diff
@@ -1365,3 +1362,97 @@ def check_fursuit_active(id_server):
...
+		killer_isEnemy = False
+		killer_isUser = False
+
+		# Generate list of enemy inflicted traumas
+		enemy_trauma_list = []
+		for atype in ewcfg.enemy_attack_type_list:
+			enemy_trauma_list.append(atype.id_type)
+
+		# Enemies have unique traumas, meaning if they have enemy trauma the killer was an enemy
+		if(user_data.trauma in enemy_trauma_list):
```
Comment (magicksRampage) [inline ewutils.py]: this is such an ass backwards way of checking this. can't we just check the cause of death instead?

File: ewwep.py [hunk 3]
```diff
@@ -1156,17 +1151,13 @@ def canAttack(cmd):
...
 		else:
 			# This should never happen. We handled all the role cases. Just in case.
 			response = "\*click* Alas, your gun has jammed."
 
 	# Send the response to the player.
-	await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, response))
-	if deathreport != "":
-		sewerchannel = ewutils.get_channel(cmd.message.server, ewcfg.channel_sewers)
-		await ewutils.send_message(cmd.client, sewerchannel, deathreport)
+	resp_cont.add_channel_response(cmd.message.channel, ewutils.formatMessage(cmd.message.author, response)) # TODO - fix this, cmd.message.channel returns none?
```
Comment (magicksRampage) [inline ewwep.py]: it should be cmd.message.channel.name

File: ew.py [hunk 4]
```diff
@@ -227,10 +255,23 @@ def die(self, cause = None):
...
 
+		self.persist()
+
+		if cause not in explosion_block_list: # Run explosion after location/stat reset, to prevent looping onto self
+			if user_hasCombustion:
+				explode_resp = "\n{} spontaneously combusts, horribly dying in a fiery explosion of slime and shrapnel!! Oh, the humanity!\n".format(server.get_member(self.id_user).display_name)
+				ewutils.logMsg("")
+				resp_cont.add_channel_response(explode_poi_channel, explode_resp)
+
+				explosion = ewutils.explode(damage = explode_damage, district_data = explode_district)
```
Comment (magicksRampage) [inline ew.py:266]: we need to remove the explode() call from explode()

**Review by magicksRampage — CHANGES_REQUESTED** [review]

File: ewcmd.py (hunk 1 above)
Comment (Froggg) [reply ewcmd.py]: alright, was just mimicking what was there before, my bad, removed check

File: ewutils.py (hunk 2 above)
Comment (Froggg) [reply ewutils.py]: yup. sure can. Guess I was thinking too much on that one

File: ewwep.py (hunk 3 above)
Comment (Froggg) [reply ewwep.py]: ah, alright

File: ew.py (hunk 4 above)
Comment (Froggg) [reply ew.py:266]: I agree, forgot to redo that change after i accidentally lost the whole project

**Pull request merged**